### ISSUE TITLE
Creating typed Client from supabase openapi schema

### ISSUE THREAD

**Issue by NixBiks** [supabase-community/supabase-py#584]
**Is your feature request related to a problem? Please describe.**

Currently the python client isn't typed based on the data model. It makes a bit "wild wild west". Using the generated OpenAPI schema it should be possible to generate a typed blient.

**Describe the solution you'd like**

I think it would be great to have similar support in Python as there is in Typescript currently. On the Typescript side of things, it is possible to generate a [typed Client](https://supabase.com/docs/reference/javascript/typescript-support)

**Additional context**
It might be useful to get some inspiration from [`openapi-python-client`](https://github.com/openapi-generators/openapi-python-client)

**Comment by anand2312** [comment]
Related: supabase-community/postgrest-py#301

This feature has been on my mind for sometime, but the DX won't be as good as it is with typescript just due to limitations of the Python type system (for example, selecting only a few columns would be almost impossible to reflect in the returned type)
It'd still be very useful to have this.

**Comment by NixBiks** [comment]
Ah yeah that's true. Typescript is definitely more powerful than the Python type system. Didn't think of all those dynamic select statements.

**Comment by NixBiks** [comment]
I went with a different solution for my use case. Since python runs server side then connecting directly to the database is often a viable solution (instead of using postgrest api). 

So I went with [Prisma Client Python](https://prisma-client-py.readthedocs.io/en/stable/). Ie. I used Prisma CLI to generate `prisma.schema` based on the existing database. Then I generated my prisma python client from the schema. And voila you have a fully typed python client to interact with your database. 

It's probably a different scope than the scope of `supabase-py` but it would still be useful for a similar supabase python solution. I'm stating it here if others are looking for a fully typed solution as well and for possible inspiration for supabase-py too.
